src/main/java/core/basesyntax/service/impl/MapTransactionService.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.TransactionType;
+import core.basesyntax.service.TransactionHandler;
+import core.basesyntax.service.TransactionService;
+import java.util.Map;
+
+public class MapTransactionService implements TransactionService {
+    private final Map<TransactionType, TransactionHandler> handlers;
+
+    public MapTransactionService(Map<TransactionType, TransactionHandler> handlers) {
+        this.handlers = Map.copyOf(handlers);
```

Constructor does not validate the `handlers` argument. The generalized requirements require null handling and informative errors: "Null handling: validate method arguments (e.g., file paths, lists) are not null and throw informative RuntimeException if violated." Add a null-check (and consider validating non-empty if appropriate) and throw a descriptive RuntimeException referencing the problem (e.g., "handlers map must not be null") so callers fail fast with a clear message.